website/source/docs/internals/providers/resource-data.html.md

```diff
@@ -0,0 +1,180 @@
+type.
+
+`ResourceData` also has a `GetOk` method that functions identically to its
+`Get` method, but with an extra return parameter. The new return parameter
+returns `true` if the property is set to a non-zero value, but with caveats.
+This is where it's important to remember that `ResourceData` is an amalgamation
+of input sources; providers cannot determine whether the property is set in the
+config, in the state, in the diff, or by using `ResourceData.Set`. The only
+information providers have available to them is that the property has been set
+at some point, and what its value is right now.
+
+## Detecting Changes
+
+When there's a difference between what our infrastructure is and what we want
+it to be, we want to be able to see what changed and what it should be. To aid
+in this, `ResourceData` provides a `HasChange` method and a `GetChange` method.
+Each takes a property key, just like `Get`. `HasChange` returns `true` if
+there's a change to that property; the change could be from drift (someone or
+something modifying the infrastructure outside of Terraform) or from config
+changes. The `GetChange` method returns two values; a basic understanding is
+that the first value is what the property _used to be_, and the second value is
+what the property _should be changed to_.  A slightly more nuanced
+understanding is that the first value is what is in the state (representing the
+state of the infrastructure as it exists) and the second value is what is in
+the config (representing what the user wants the infrastructure to be).
+
+You'll notice there's no way to tell whether the state changed or the config
+changed; this is a common misconception. Terraform never diffs states or
+configs, it only ever diffs what is and what should be.
+
+Be sure to call `HasChange` before calling `GetChange`; there are some cases
+where `GetChange` would return both values as equal, but reflect a change. This
+can happen, for example, when a boolean property has a default value of `false`
+and was not specified, then gets set explicitly to `false`.
+
+`GetChange` is only really necessary when you need to know what the previous
+value of the field was; if you only need the new value, `Get` is sufficient.
+`GetChange` is commonly used in cases where the API requires you to explicitly
+remove and add items—for example, when there's a change in tags and the API
+only offers `AddTag` & `RemoveTag` methods without a way to just change all
+tags at once. In that case, you need to know what the previous tag value was,
+so it can be removed.
+
+## Setting State
+
+At the end of your `Create`, `Get`, `Update`, and `Delete` functions, whatever
+is in your `ResourceData` object is set as state. To manipulate this, the `Set`
+method is provided. It takes the key of a property, just like `Get`, as an
+argument, and the value to set it to.
+
+It's important that all your properties get set using the `Set` method, as
```

Comment:
"get set using the `Set` method". Please try to rephrase this using words that aren't also keywords in this context.